homeassistant/components/systemmonitor/repairs.py

```diff
@@ -0,0 +1,67 @@
+"""Repairs platform for the System Monitor integration."""
+
+from __future__ import annotations
+
+from typing import Any, cast
+
+from homeassistant import data_entry_flow
+from homeassistant.components.binary_sensor import DOMAIN as BINARY_SENSOR_DOMAIN
+from homeassistant.components.repairs import ConfirmRepairFlow, RepairsFlow
+from homeassistant.components.sensor import DOMAIN as SENSOR_DOMAIN
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+
+
+class ProcessFixFlow(RepairsFlow):
+    """Handler for an issue fixing flow."""
+
+    def __init__(self, entry: ConfigEntry, processes: list[str]) -> None:
+        """Create flow."""
+        self.entry = entry
+        self._processes = processes
+        super().__init__()
+
+    async def async_step_init(
+        self, user_input: dict[str, str] | None = None
+    ) -> data_entry_flow.FlowResult:
+        """Handle the first step of a fix flow."""
+        return await self.async_step_migrate_process_sensor()
+
+    async def async_step_migrate_process_sensor(
+        self, user_input: dict[str, Any] | None = None
+    ) -> data_entry_flow.FlowResult:
+        """Handle the options step of a fix flow."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="migrate_process_sensor",
+                description_placeholders={"processes": ", ".join(self._processes)},
+            )
+        options = dict(self.entry.options)
+        resources: list[str] | None = options.get("resources")
+        processes: dict[str, list[str]] | None = options.get(SENSOR_DOMAIN)
+        new_options: dict[str, Any] = {}
+        if processes:
+            new_options[BINARY_SENSOR_DOMAIN] = processes
```

Comment:
Why do we modify the config entry options here when we have the config entry migration? We should just reload the config entry in this flow to have it migrate, I think.

Reply:
The migrations makes a copy of the sensors and add them as binary sensors so we have both.
The repair removes the sensor ones so we only have binary sensors left.

Could make it more obvious in the repair flow by popping the sensors instead of constructing a new option dict?

Reply:
Ok.

Should we remove the entity registry entries for the process sensors too?